starter_code/src/movies.js

```diff
@@ -1,15 +1,148 @@
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(movies) {
+   
+    let newMovies = [...movies];
+    if ( movies.length == 0  ){
+       return newMovies ;
+    }
+    
+    newMovies.sort(function(moviea, movieb) { 
+        return moviea.year - movieb.year  ;
+    });
+    
+    newMovies.sort((a, b) => (a.year > b.year) ? 1 : (a.year === b.year) ? ((a.title > b.title) ? 1 : -1) : -1 )
+     
+    //console.log(newMovies); 
+    return newMovies;    
+}    
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+    function howManyMovies(movies) {
 
+        if (movies.length == 0){
+            return 0;
+        }      
+
+        const moviesNumbers = movies.filter(movie => movie.director == "Steven Spielberg" );
+
+        const moviesNumbers2 = moviesNumbers.filter(movie2 => movie2.genre.indexOf('Drama') > -1 );
+
+        return moviesNumbers2.length; 
+
+    } 
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+    function orderAlphabetically(movies) {
+        
+        let newMovies = [...movies];
+        newMovies.sort((a,b) =>{
+            if (a.title.toLowerCase() > b.title.toLowerCase() ) return 1;   
+            if (a.title.toLowerCase() < b.title.toLowerCase() ) return -1;
+            return 0;
+        })
+        
+        let moviesTitle = [];
+        newMovies.forEach(element => moviesTitle.push(element.title));
+        
+        return moviesTitle.slice(0, 20) ;
+    };
```

Comment:
Here since you had to transform the original array and get only the titles, you could have used map for it which creates a new copy of the array.

Then sort it and then slice it.